packages/ui/src/primitives/tailwind/Color/index.tsx

```diff
@@ -52,36 +52,68 @@ export function ColorInput({
 }: ColorInputProp) {
   const color = new Color(value)
   const hexColor = '#' + color.getHexString()
+  const [isPickerOpen, setIsPickerOpen] = useState(false)
+  const pickerRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLDivElement>(null)
+
+  const handleTogglePicker = () => {
+    setIsPickerOpen(!isPickerOpen) // Toggle visibility
+    onRelease && onRelease(color)
+  }
 
   const handleChange = (result: ColorResult) => {
     const color = new Color(result.hex)
     onChange(color)
   }
+
+  // Close the picker if clicking outside the input or picker - this method didn't require a new signature
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (
+        inputRef.current &&
+        !inputRef.current.contains(event.target) &&
+        pickerRef.current &&
+        !pickerRef.current.contains(event.target)
+      ) {
+        setIsPickerOpen(false)
+        if (onRelease) onRelease(color)
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
```

Comment:
can we use the client input system to handle this ?

Reply:
I don't think so, no